lib/index-web.js

```diff
@@ -133,7 +133,11 @@ module.exports = function(config, auth, storage) {
     var getData = function(i) {
       storage.get_package(results[i].ref, function(err, entry) {
         if (!err && entry) {
-          packages.push(entry.versions[entry['dist-tags'].latest])
+          auth.allow_access(entry.name, req.remote_user, function(err, allowed) { // TODO: This may cause performance issue?
```

Comment:
Reply of https://github.com/verdaccio/verdaccio/pull/161#discussion_r112496123
@Meeeeow  
About the `can('access')`, yeah you are right. No way to use the middleware. Perhaps you update `allowed` as `is_allowed`, to be consistent with the middleware.
About the performance, I haven't test it with thousands of packages, it worth to try in the future and see what's happen.